test/unit/math/prim/mat/vectorize/expect_std_vector_error.hpp

```diff
@@ -0,0 +1,24 @@
+#ifndef TEST_UNIT_MATH_PRIM_MAT_VECTORIZE_EXPECT_STD_VECTOR_ERROR_HPP
+#define TEST_UNIT_MATH_PRIM_MAT_VECTORIZE_EXPECT_STD_VECTOR_ERROR_HPP
+
+#include <gtest/gtest.h>
+#include <stdexcept>
+#include <vector>
+
+template <typename F, typename T>
+void expect_std_vector_error() {
+  using std::vector;
+  vector<double> invalid_inputs = F::invalid_inputs();
+  if (invalid_inputs.size() > 0) {
```

Comment:
@rayleigh, I would prefer if we did this before merging.

(I also recommend _Refactoring_.)